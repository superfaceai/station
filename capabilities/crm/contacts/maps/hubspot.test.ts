import { RecordingScopes, SuperfaceTest } from '@superfaceai/testing-lib';

function afterRecordingLoad(scopes: RecordingScopes) {
  scopes.forEach(scope => {
    scope.filteringPath(
      /hapikey=[^&]*/g,
      'hapikey=credentials-removed-to-keep-them-secure'
    );
  });
}

describe(`crm/contacts/hubspot`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('Create', () => {
    it('should perform successfully', async () => {
      await expect(
        superface.run(
          {
            profile: 'crm/contacts',
            provider: 'hubspot',
            useCase: 'Create',
            input: {
              email: 'test@example.com',
              firstName: 'Test',
              lastName: 'User',
              company: 'Station Test',
              country: 'USA',
              customProperties: {
                myproperty: 'value',
              },
            },
          },
          {
            afterRecordingLoad,
          }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('Update', () => {
    it('should return error if id is missing', async () => {
      await expect(
        superface.run(
          {
            profile: 'crm/contacts',
            provider: 'hubspot',
            useCase: 'Update',
            input: {
              email: 'test@example.com',
            },
          },
          {
            afterRecordingLoad,
          }
        )
      ).resolves.toMatchSnapshot();
    });

    it('should perform successfully', async () => {
      await expect(
        superface.run(
          {
            profile: 'crm/contacts',
            provider: 'hubspot',
            useCase: 'Update',
            input: {
              id: '401',
              firstName: 'Updated first name',
            },
          },
          {
            afterRecordingLoad,
          }
        )
      ).resolves.toMatchSnapshot();
    });
  });

  describe('Search', () => {
    it('should return error for unknown operator', async () => {
      await expect(
        superface.run(
          {
            profile: 'crm/contacts',
            provider: 'hubspot',
            useCase: 'Search',
            input: {
              property: 'firstname',
              operator: '%',
              value: 'John',
            },
          },
          {
            afterRecordingLoad,
          }
        )
      ).resolves.toMatchSnapshot();
    });

    it('should perform successfully', async () => {
      await expect(
        superface.run(
          {
            profile: 'crm/contacts',
            provider: 'hubspot',
            useCase: 'Search',
            input: {
              property: 'firstname',
              operator: 'EQ',
              value: 'Test',
            },
          },
          {
            afterRecordingLoad,
          }
        )
      ).resolves.toMatchSnapshot();
    });
  });
});
