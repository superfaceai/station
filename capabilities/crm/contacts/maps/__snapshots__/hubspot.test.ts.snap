// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crm/contacts/hubspot Create should perform successfully 1`] = `
Object {
  "value": Object {
    "id": "601",
  },
}
`;

exports[`crm/contacts/hubspot Search should perform successfully 1`] = `
Object {
  "value": Array [
    Object {
      "company": undefined,
      "country": undefined,
      "email": "test@example.com",
      "firstName": "Test",
      "id": "601",
      "lastName": "User",
      "phone": undefined,
    },
  ],
}
`;

exports[`crm/contacts/hubspot Search should return error for unknown operator 1`] = `
Object {
  "error": "Input validation failed:
Path: input.operator
Error: Invalid enum value: \\"%\\"
Path: input.operator
Error: Invalid enum value: \\"%\\"",
}
`;

exports[`crm/contacts/hubspot Update should perform successfully 1`] = `
Object {
  "value": Object {
    "id": "401",
  },
}
`;

exports[`crm/contacts/hubspot Update should return error if id is missing 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[1].statements[0]
Original Map Location: Line 52, column 3",
}
`;
